### ISSUE TITLE
API Export returning 403 on quota endpoint

### ISSUE THREAD

**Issue by renankaic** [Axway-API-Management-Plus/apim-cli#303]
### APIM-CLI version

1.12.2

### API-Management version

7.7.20210530

### Bug description

Hello.
I'm trying to export of my frontend APIs to automate a backup in my environment.

When using the following command it works as it should and it lists all the APIS:
`apim api get -s qa`

But, when trying to export all of them to JSON:
`apim api get -s qa --output json -t /home/axway/exports/ -useFEAPIDefinition `

It gives the following error:
`Error cant load subscribes applications from API-Manager. Can't parse response: <html><body>Forbidden</body></html>
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"<html><body>Forbidden</body></html>"; line: 1, column: 2]`

### Steps to reproduce

1 - Configure APIM CLI
2 - Configure env.qa.settings with proper host, username and password
3 - Execute "`apim api get -s qa --output json -t [target-directory] -useFEAPIDefinition `" command
4 - Receive the above error

### Relevant log output

```shell
/usr/bin/java

----------------------------------------------------------------------------------------
API-Manager CLI: 1.12.2
                                                                        
To report issues or get help, please visit: 
https://github.com/Axway-API-Management-Plus/apim-cli
----------------------------------------------------------------------------------------
Module: API - E X P O R T / U T I L S (1.12.2)
----------------------------------------------------------------------------------------
2215   INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20210530) on: https://bp1haebanmdev:8075
2748   INFO  APIManagerAPIAdapter| Loading application quotas for 4 subscribed applications.
2855   INFO  APIManagerAPIAdapter| Successfully downloaded API-Specification with version 3.0 from Frontend-API.

Loading details of 33 APIs [----------] 0%3145   INFO  APIManagerAPIAdapter| Loading application quotas for 8 subscribed applications.
3379   INFO  APIManagerAPIAdapter| Successfully downloaded API-Specification with version 2.0 from Frontend-API.
3436   INFO  agger2xSpecification| Used the backendBasepath: 'http://[our-ip]:[our-port]/sap/opu/odata/sap/ZGS_TESTDRIVE_SRV' to adjust the API-Specification.

[...]
[...]
[...]
Loading details of 33 APIs [*********-] 90%10370  ERROR    APIMgrAppsAdapter| Error cant load subscribes applications from API-Manager. Can't parse response: <html><body>Forbidden</body></html>
com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"<html><body>Forbidden</body></html>"; line: 1, column: 2]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[jackson-core-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:735) ~[jackson-core-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:659) ~[jackson-core-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2005) ~[jackson-core-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:802) ~[jackson-core-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4761) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4667) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3612) ~[jackson-databind-2.13.3.jar:2.13.3]
	at com.axway.apim.adapter.clientApps.APIMgrAppsAdapter.getAppsSubscribedWithAPI(APIMgrAppsAdapter.java:177) ~[apimcli-apim-adapter-1.12.2.jar:1.12.2]
	at com.axway.apim.adapter.apis.APIManagerAPIAdapter.addClientApplications(APIManagerAPIAdapter.java:460) ~[apimcli-apim-adapter-1.12.2.jar:1.12.2]
	at com.axway.apim.adapter.apis.APIManagerAPIAdapter.getAPIs(APIManagerAPIAdapter.java:103) ~[apimcli-apim-adapter-1.12.2.jar:1.12.2]
	at com.axway.apim.APIExportApp.execute(APIExportApp.java:212) ~[apimcli-apis-1.12.2.jar:1.12.2]
	at com.axway.apim.APIExportApp.exportAPI(APIExportApp.java:87) ~[apimcli-apis-1.12.2.jar:1.12.2]
	at com.axway.apim.APIExportApp.exportAPI(APIExportApp.java:78) ~[apimcli-apis-1.12.2.jar:1.12.2]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_332]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_332]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_332]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_332]
	at com.axway.apim.cli.APIManagerCLI.run(APIManagerCLI.java:141) ~[apimcli-tool-1.12.2.jar:1.12.2]
	at com.axway.apim.cli.APIManagerCLI.main(APIManagerCLI.java:77) ~[apimcli-tool-1.12.2.jar:1.12.2]
10377  INFO          AppException| You may enable debug to get more details. See: https://github.com/Axway-API-Management-Plus/apim-cli/wiki/9.1.-Enable-Debug
10377  ERROR         AppException| Cannot read APIs from API-Manager
                                 | Error cant load subscribes applications from API-Manager.
```

**Comment by JSomervilleFromAz** [comment]
How api-cli deals with this scenario may need a look.

But, note that the "Unexpected character ('<'" error itself may just be that an SOAP/XML based API is being accessed as JSON.
That part of the error could also be a logging artifact; see 7th comment down on internal Jira RDAPI-25253, about how different parsers may be tried in turn.

**Comment by renankaic** [comment]
> How api-cli deals with this scenario may need a look.
> 
> But, note that the "Unexpected character ('<'" error itself may just be that an SOAP/XML based API is being accessed as JSON. That part of the error could also be a logging artifact; see 7th comment down on internal Jira RDAPI-25253, about how different parsers may be tried in turn.

Yes! It is true! Our API Gateway return a SOAP response when an error happen. I don't know yet why this happens, but I'll try to figure it out by continuing the documentation reading.

**Comment by rathnapandi** [comment]
Hi @renankaic, 

To debug the issue, can you download version 1.12.3, add command line parameter -disableCompression and JVM parameters in apim.sh? which will print all request and responses. 

-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.SimpleLog
-Dorg.apache.commons.logging.simplelog.showdatetime=true
-Dorg.apache.commons.logging.simplelog.log.org.apache.http=DEBUG

**Comment by renankaic** [comment]
> Hi @renankaic,
> 
> To debug the issue, can you download version 1.12.3, add command line parameter -disableCompression and JVM parameters in apim.sh? which will print all request and responses.
> 
> -Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.SimpleLog -Dorg.apache.commons.logging.simplelog.showdatetime=true -Dorg.apache.commons.logging.simplelog.log.org.apache.http=DEBUG



Hello my friend @rathnapandi . Sorry for my late in answering, my work in these days were heavy.

I did as you suggested, but I don't know if it worked.

I put the JVM parameters in apim.sh with the following content:
`"$_java" -Xms64m -Xmx256m -Dlog4j.configurationFile=../lib/log4j2.xml -Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.SimpleLog -Dorg.apache.commons.logging.simplelog.showdatetime=true -Dorg.apache.commons.logging.simplelog.log.org.apache.http=DEBUG -classpath "$CP" com.axway.apim.cli.APIManagerCLI "${@}"
`

Then, changed the APIM_CLI_HOME

Then I executed the following command:
`apim.sh api get -s qa --output json -t /home/axway/exports/ -useFEAPIDefinition -disableCompression  > logs.log`

Then, a very big log started to appear in the shell but for some reason it doesn't write to `logs.log` file as I stated above. It just writes the same log that I put in the start of this issue.

How can I provide the full log to you?

Thank you!
